packages/components/src/ui/create-styles/create-style-system/create-styled-components.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { INTERPOLATION_CLASS_NAME } from './constants';
+import { tags } from './tags';
+import { compileInterpolatedStyles, getInterpolatedClassName } from './utils';
+
+/**
+ * @typedef CreateStyledComponentsProps
+ * @property {import('../create-compiler').Compiler} compiler The (custom) Emotion instance.
+ * @property {import('./polymorphic-component').CoreElements} core The collection of coreElements.
+ */
+
+/**
+ * Creates a set of styled components for the Style system.
+ * These styled components are similarly to Emotion's or Styled-Components styled.div``.
+ *
+ * A big difference is that the Style system's styled components do NOT require
+ * context connection at all. This is HUGE for performance as there are far less
+ * React.Component nodes within the render tree.
+ *
+ * This is thanks to how the Style system compiles and coordinates style values.
```

Comment:
This sound like these are "shortcuts" to using raw `styled` right? Are we using these? do we need them?

Reply:
I'm not following your question. What do you mean by a "shortcut" and "raw `styled`"?

Reply:
Nevermind :)